addon/scheduler.ts

```diff
@@ -52,16 +91,30 @@ export function beginTransition(): void {
 }
 
 export function endTransition(): void {
-  _whenRouteDidChange.resolve();
+  if (CAPABILITIES.performanceObserverEnabled) {
+    if (!schedulerOptions.emitMarks) {
+      return;
+    }
+
+    _emitMarks();
+  } else {
+    _whenRouteDidChange.resolve();
+  }
 }
 
-export function setupRouter(router: Router): void {
+export function setupRouter(router: Router, options: SchedulerOptions): void {
   if (IS_FASTBOOT || (router as any)[APP_SCHEDULER_HAS_SETUP]) {
     return;
   }
 
   (router as any)[APP_SCHEDULER_HAS_SETUP] = true;
 
+  if (CAPABILITIES.performanceObserverEnabled) {
+    schedulerOptions = assign(DEFAULT_OPTIONS, options);
+
+    installObserver();
```

Comment:
Can we make it so that we only install the observer once (ever)?

Reply:
Yep. We spoke about this, I just forgot. Will fix that now.